specification/dartLangSpec.tex

```diff
@@ -3073,20 +3076,32 @@ \subsection{Constructors}
 where $X_j, j \in 1 .. m$ are the formal type parameters of $C$
 and $T_j, j \in 1 .. m$ are specified in the given context.
 We may also omit such a substitution when the given context is
-the instance scope of $C$, where $X_1, \ldots, X_m$ are in scope.
+the class scope of $C$, where $X_1, \ldots, X_m$ are in scope.
 }
 
-\commentary{
+\commentary{%
 A constructor declaration may conflict with static member declarations
 (\ref{classMemberConflicts}).
 }
 
-% In what scope do constructors go? The simple names of named constructors go in the static scope of the class. Unnamed ones go nowhere, but we use the class name to refer to them; the class name could also in the static scope of the class as well to prevent weird errors, or we could ban it explicitly and avoiding duplication. Similarly, the instance scope could contain the constructor names and class name, or we could have special rules to prevent collisions between instance members and constructors or the class.
-
-% The enclosing scope of a generative constructor is the instance scope of the class in which it is declared (but what about redirecting?)
+\commentary{%
+A constructor declaration does not introduce a name into a scope.
```

Comment:
(Not sure that was a good idea, but it is what it is).

Reply:
Yes, I think this is a rule which is consistent with the actual treatment of constructors that we've had all the time. I don't think it's particularly problematic, but it might need to be considered an exception by 'dart:mirrors'.

Reply:
Not sure it affects `dart:mirrors`. Mirrors are about declarations, not scopes. 
(I see "scopes" as lexical constructs used to look up single identifiers. Since constructors have names that are not single identifiers, it makes sense that they are not in any scope. We also do non-scope lookup on static accesses like `Function.apply`, we look for static members in the declaration of the *class*, not names in a scope introduced by the class. The exception is import prefix "scopes", which are named scopes accessed by a qualified identifier instead of a single identifier).

Reply:
I agree on those things.